File: backend/src/main/java/com/example/backend/Endpoints/dto/ProductDto.java
package com.example.backend.Endpoints.dto;

import io.soabase.recordbuilder.core.RecordBuilder;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Builder;

@RecordBuilder
public record ProductDto(
        Long id,
        @NotBlank(message = "Product name is required")
        @Size(max = 100, message = "Product name cannot exceed 100 characters")
        String name,
        @Size(max = 100, message = "Product description cannot exceed 500 characters")
        String description,
        @NotNull(message = "Product price is required")
        Double price,
        boolean sold,
        UserDetailDto user,
        byte[] imageData
) {
}

File: frontend/src/app/dtos/productDto.ts
import {UserDetail} from "./auth-request";

export class ProductDto {
  constructor(
    public id: number,
    public name: string,
    public description: string,
    public sold:boolean,
    public price: number,
    public user: UserDetail,
    public imageData: string,
  ) {
  }
}

export class ProductSearchDto {
  constructor(
    public name: string,
    public description: string,
    public priceFrom: number,
    public priceTo: number,
  ) {
  }
}
